modules/transport-netty4/build.gradle

```diff
@@ -38,6 +38,9 @@ apply plugin: 'opensearch.yaml-rest-test'
 apply plugin: 'opensearch.java-rest-test'
 apply plugin: 'opensearch.internal-cluster-test'
 
+// The transport-netty4 plugin is published to maven
+apply plugin: 'opensearch.publish'
```

Comment:
@dbwiddis : Thanks for this PR. I have few questions around this change for my own understanding. 
1. How is opensearch-sdk consuming this plugin ? 
2. Why do we want to have plugin maintained in core and consumed in opensearch-sdk ?

Reply:
> How is opensearch-sdk consuming this plugin ?

We are using the `TransportService` wrapping `Netty4Transport` as our means of having the extensions communicate with OpenSearch itself.  See [here](https://github.com/opensearch-project/opensearch-sdk/blob/f9b3157af6b6b8cf05d4fb34ce6a8f529446a55f/src/main/java/org/opensearch/sdk/ExtensionsRunner.java#L145-L175) for example.

Presently we have simply copied all the classes over into a [package in the sdk](https://github.com/opensearch-project/opensearch-sdk/tree/main/src/main/java/org/opensearch/sdk/netty4) during development. We'd like to move this to a dependency published to Maven.

> Why do we want to have plugin in core but consumed in opensearch-sdk ?

It will also be consumed in core for remote communication with the extensions. That has not yet been added as we are temporarily using plugin features. See [here](https://github.com/opensearch-project/OpenSearch/blob/be1a605479a8da1809e6d706abbd0b0e59dea16f/server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java#L243) for where the TransportService is used and will eventually also consume this dependency.

@owaiskazi19 please clarify/correct anything I may have misstated.

Reply:
Thank you @dbwiddis for the details.